service/history/queues/fifo_scheduler.go

```diff
@@ -47,52 +41,45 @@ type (
 	// and always schedule tasks in fifo order regardless
 	// which namespace the task belongs to.
 	FIFOScheduler struct {
-		wRRScheduler tasks.Scheduler[Executable]
+		scheduler tasks.Scheduler[Executable]
```

Comment:
what is the difference of this FIFOScheduler and the one under common/tasks/fifo.scheduler.go?

Reply:
There's no difference actually. But the Scheduler interface defined in `history/queues` package is different from the one in `common/tasks`, so need an adapter. 

Re. the actual difference of the interface, please check this [PR](https://github.com/temporalio/temporal/pull/3266/files#diff-36d52beeb8ce1d18e54aa8aefb9352461e6245bc0d693b397d80ea829f4ac4cfR61). The definition here needs to be updated, and I will update this PR after the linked one is landed.

Reply:
Updated. This file is now gone. 

The scheduler defined in `history/queues` package is a thin wrapper around the one defined in `common/tasks`, to support getting the func for task chan and weight assignment.